pyrate/core/gamma.py

```diff
@@ -230,24 +221,34 @@ def get_header_paths(input_file, slc_dir):
     :rtype: list
     """
     _, file_name = split(input_file)
-    matches = PTN.findall(file_name)
-    headers = [glob2.glob(join(slc_dir, '**/*%s*slc.par' % m))[0]
-            	for m in matches]
-    return headers
-
+    epochs = PTN.findall(file_name)
+    header_names = cf.parse_namelist(slc_file_list)
+    matches = [hdr for hdr in header_names if any(e in hdr for e in epochs)]
```

Comment:
An error is thrown at some point if an epoch is not found in the header list right?
```
# along the lines of :
if not any(e in epochs for e in hdr ):
    Exception("header not available for an epoch")
```